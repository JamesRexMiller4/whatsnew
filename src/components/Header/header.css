header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    margin-bottom: 20px;
}
h1 {
    color: aquamarine;
}

.search-bar-input {
    width: 30%;
    height: 40px;
    font-size: 20px;
    font-weight: bold;
    border-radius: 5px;
}

.search-bar-btn {
    width: 10%;
    font-size: 20px;
    font-weight: bold;
    border-radius: 20px;
    background-color: royalblue;
}

.search-bar-btn:hover {
    background-color: rebeccapurple;
    color: white;
}