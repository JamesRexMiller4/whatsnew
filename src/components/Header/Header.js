import React, {Component} from 'react';
import './header.css';

class Header extends Component {
    constructor() {
        super();
        this.state = {
            query: ''
        }
    }

    handleChange = event => {
        this.setState({query: event.target.value})
    }

    submitQuery = () => {
        const query = this.state
        this.props.filterNews(query)
        document.querySelector('#search-bar').value = ''
        this.setState({query: ''})
    }


    render() {
        return (
            <header>
                <form>
                <h1>What's New?</h1>
                <input type='text' id='search-bar' onChange={event => this.handleChange(event)} className='search-bar-input' placeholder='Search for news articles here'/> 
                <button type='button' className='search-bar-btn' onClick={event => this.submitQuery()}>Search Now</button>
                </form>
            </header>
        )
    }
}

export default Header