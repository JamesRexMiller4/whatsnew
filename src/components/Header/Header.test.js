import React from 'react';
import Header from './Header';
import {shallow} from 'enzyme';

describe('Header', () => {

    let wrapper, mockFilterNews;

    beforeEach(() => {
        mockFilterNews = jest.fn()
        wrapper = shallow(<Header filterNews = {mockFilterNews}/>)
        wrapper.setState({query: ''})
    })

    it('should match the snapshot with all the data passed correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should have its state assigned to the appropriate data', () => {
        expect(wrapper.state('query')).toEqual('')
    })

    describe('Methods', () => {

        it('should be able set the components state after an onChange event', () => {
            const mockEvent = {target: {value: 'NASA'}}
            const expected = 'NASA'
            wrapper.instance().handleChange(mockEvent)
            expect(wrapper.state('query')).toEqual(expected)
        })

        it('should trigger the filterNews method on a click event, and reset the state of the header component', () => {
            const mockQuery = {query: 'NASA'}
            const mockSubmitQuery = jest.fn()
            wrapper.setState({query: 'NASA'})
            expect(wrapper.state('query')).toEqual('NASA')
            wrapper.find('.search-bar-btn').simulate('click')
            expect(mockSubmitQuery).toHaveBeenCalled()
            expect(mockFilterNews).toHaveBeenCalledWith(mockQuery)
        })
    })
})