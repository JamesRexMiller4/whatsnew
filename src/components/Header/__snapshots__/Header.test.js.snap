// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all the data passed correctly 1`] = `
<header>
  <form>
    <h1>
      What's New?
    </h1>
    <input
      className="search-bar-input"
      id="search-bar"
      onChange={[Function]}
      placeholder="Search for news articles here"
      type="text"
    />
    <button
      className="search-bar-btn"
      onClick={[Function]}
      type="button"
    >
      Search Now
    </button>
  </form>
</header>
`;
